lib/internal/test_runner/fake_timers.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const {
+  DateNow,
+  SafeMap,
+  Symbol,
+  globalThis,
+} = primordials;
+
+class Timers {
+  constructor() {
+    this.timers = new SafeMap();
```

Comment:
I appreciate the implementation using the same approach as Sinon's, but Node can do (much) better in terms of performance if we used our (already existing!) priority queue implementation instead of the object. 

This is a pain point in sinon and we've tried a few times to swap (which was good enough but there were backwards computability issues)

Reply:
I'm happy that it's the same approach because I was just typing it 😂

Do you have  an example for this priority queue impl?

Reply:
If you look at timers.js it uses a priority queue. The implementation is inhttps://github.com/nodejs/node/blob/d225d957586198ca6e7c63150ef8354bfb7141b9/lib/internal/priority_queue.js#L7 and you can see a call site in timers.js

Reply:
just added it!